client/src/api/block_builder/transaction.rs

```diff
@@ -146,3 +161,41 @@ pub fn verify_semantic(
 
     semantic_validation(context, inputs.as_slice(), transaction.unlocks()).map_err(Error::BlockError)
 }
+
+/// Verifies that the transaction payload doesn't exceed the block size limit with 8 parents.
+pub fn validate_transaction_payload_length(transaction_payload: &TransactionPayload) -> Result<()> {
+    let transaction_payload_bytes = transaction_payload.pack_to_vec();
+    if transaction_payload_bytes.len() > MAX_TX_LENGTH_FOR_BLOCK_WITH_8_PARENTS {
+        return Err(Error::InvalidTransactionPayloadLength {
+            length: transaction_payload_bytes.len(),
+            max_length: MAX_TX_LENGTH_FOR_BLOCK_WITH_8_PARENTS,
+        });
+    }
+    Ok(())
+}
+
+/// Verifies that the transaction essence doesn't exceed the block size limit with 8 parents.
+/// Assuming one signature unlock and otherwise reference/alias/nft unlocks. `validate_transaction_payload_length()`
+/// should later be used to check the length again with the correct unlocks.
+pub fn validate_regular_transaction_essence_length(
+    regular_transaction_essence: &RegularTransactionEssence,
+) -> Result<()> {
+    let regular_transaction_essence_bytes = regular_transaction_essence.pack_to_vec();
+
+    // Assuming there is only 1 signature unlock and the rest is reference/alias/nft unlocks
+    let reference_alias_nft_unlocks_amount = regular_transaction_essence.inputs().len() - 1;
+
+    // Max tx payload length - length for one signature unlock (there might be more unlocks, we check with them
+    // later again, when we built the transaction payload)
+    if regular_transaction_essence_bytes.len()
+        > MAX_TX_LENGTH_FOR_BLOCK_WITH_8_PARENTS
+            - SINGLE_UNLOCK_LENGTH
+            - (reference_alias_nft_unlocks_amount * REFERENCE_ALIAS_NFT_UNLOCK_LENGTH)
+    {
+        return Err(Error::InvalidRegularTransactionEssenceLength {
+            length: regular_transaction_essence_bytes.len(),
+            max_length: MAX_TX_LENGTH_FOR_BLOCK_WITH_8_PARENTS - SINGLE_UNLOCK_LENGTH,
```

Comment:
Shouldn't there be `- (reference_alias_nft_unlocks_amount * REFERENCE_ALIAS_NFT_UNLOCK_LENGTH)` as well?

Reply:
Yes sir